### PR TITLE
CLN/BUG: array_equivalent on nested objects

### PR DESCRIPTION
**Pull request by jbrockmendel**
5/5

The included code check won't pass until the others in this sequence are merged.

### CODE REVIEW

File: pandas/_libs/testing.pyx [hunk 1]
```diff
@@ -143,11 +143,9 @@ cpdef assert_almost_equal(a, b,
...
                 assert_attr_equal('dtype', a, b, obj=obj)
 
-            try:
-                if array_equivalent(a, b, strict_nan=True):
-                    return True
-            except:
-                pass
+            if array_equivalent(a, b, strict_nan=True):
+                return True
+
```
Comment (jbrockmendel) [inline pandas/_libs/testing.pyx]: need to look if we should return False on 148

**Comment by jbrockmendel** [comment]
reverted the code check since one of the other PRs in this sequence is in a holding pattern.  this should now be straightforward

File: pandas/_libs/testing.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/testing.pyx]: Looking at this more closely, I've decided it should be handled separately.  (also there are some optimizations to be made in that separate pass).  So this should be good to go.

**Comment by jbrockmendel** [comment]
@jreback gentle ping.  there is at least one more Exception we'll be able to get rid of once the array_equivalent bug here is fixed

**Comment by jreback** [comment]
cool thanks @jbrockmendel

**Pull request merged**